ui/src/routes/incarnations/List.tsx

```diff
@@ -31,35 +34,57 @@ const Table = styled.table(({ theme }) => ({
   }
 }))
 
+const CellText = styled.div({
+  whiteSpace: 'nowrap',
+  overflow: 'hidden',
+  textOverflow: 'ellipsis',
+  maxWidth: '100%'
+})
+
 export const IncarnationsList = () => {
+  const { search } = useToolbarSearchStore()
   const { isLoading, isError, data } = useQuery(['incarnations'], incarnations.get) // TODO: wrap it to useIncarnationsQuery
   if (isLoading) {
     return <Section>Loading...</Section>
   }
   if (isError) {
     return <Section>Error loading incarnations 😔</Section>
   }
+  const _data = data.filter(({ incarnationRepository }) => incarnationRepository.toLowerCase().includes(search.toLowerCase()))
   return (
     <Section>
       <h3>Incarnations</h3>
       <Table>
         <thead>
           <tr>
-            <th>Id</th>
-            <th>Repository</th>
-            <th>Target directory</th>
-            <th>Commit</th>
-            <th>Merge Request</th>
+            <th style={{ width: 40 }}>Id</th>
+            <th style={{ width: 'calc(50% - 40px - 218px)' }}>Repository</th>
+            <th style={{ width: 'calc(50% - 40px - 218px)' }}>Target directory</th>
+            <th style={{ width: 218 }} />
           </tr>
         </thead>
         <tbody>
-          {data.map(x => (
+          {_data.map(x => (
             <tr key={x.id}>
               <td>{x.id}</td>
-              <td>{x.incarnationRepository}</td>
+              <td>
+                <CellText>{x.incarnationRepository}</CellText>
+              </td>
               <td>{x.targetDirectory}</td>
-              <td><a href={x.commitUrl} title={x.commitUrl} target="_blank" rel="noreferrer">{x.commitUrl}</a></td>
-              <td>{x.mergeRequestUrl}</td>
+              <td>
+                <ButtonLink size="small" disabled={!x.commitUrl} href={x.commitUrl} title={x.commitUrl}>
+                  <Hug as="span" mr={4}>Commit</Hug>
+                  <Commit />
+                </ButtonLink>
+                <Hug as="span" ml={4}>
+                  <ButtonLink size="small" disabled={!x.mergeRequestUrl} href={x.mergeRequestUrl ?? undefined} title={x.mergeRequestUrl ?? undefined}>
+                    <Hug as="span" flex={['aic', 'jcsb']}>
+                      <Hug as="span" mr={4}>Merge request</Hug>
+                      <MergeRequest />
+                    </Hug>
+                  </ButtonLink>
```

Comment:
IMHO: would be nice to have the commit sha and the merge request id in the button links. WDYT?

Reply:
mmh - I'm not sure if that's very useful 😄

Reply:
I can add a tooltip with hash, this will leave table minimalistic, but will show additional info by demand